KoeBook.Core/Services/ClaudeAnalyzerService.cs

```diff
@@ -0,0 +1,235 @@
+
+                Notes:
+                - For narration parts, which are not enclosed in quotation marks, select the narrator.
+                - For dialogues enclosed in quotation marks, assign a voice other than the narrator.
+                - In the character description, include the appropriate voice characteristics.
+                Tasks: Based on the notes above, perform the following two tasks:
+                - List of characters Objective: To understand the characters appearing in the text, list the character ID, name, and description for all characters who speak at least one line.
+                - Output the speaking character or narrator for each line Objective: To identify which character is speaking in each line of the text, output the speaking character or narrator for all lines. Carefully recognize the context and output with attention. Select only one character_id per line.
+                - Revise CHARACTER LIST and VOICE ID (in the event that the CHARACTER LIST is incomplete)
+                Output Format:
+                [CHARACTER LIST]
+                c0. ナレーター: {character_and_voice_description, example:"The person who speaks the narration parts. A calm-toned male voice."}
+                c1. {character_name}: {character_and_voice_description}
+                {character_id}. {character_name}: {character_and_voice_description}
+                ...
+                [VOICE ID]
+                1. {character_id} {narration|dialogue}
+                2. {character_id} {narration|dialogue}
+                3. {character_id} {narration|dialogue}
+                ...
+                [REVISE CHARACTER LIST]
+                c0. ナレーター: {character_and_voice_description}
+                c1. {character_name}: {character_and_voice_description}
+                {character_id}. {character_name}: {character_and_voice_description}
+                ...
+                [REVISE VOICE ID]
+                1. {character_id} {narration|dialogue}
+                2. {character_id} {narration|dialogue}
+                3. {character_id} {narration|dialogue}
+                ...
+                """;
+    }
+
+    private string CreatePrompt2(List<Character> characterList)
+    {
+        StringBuilder sb = new StringBuilder();
+        foreach (var character in characterList)
+        {
+            sb.AppendLine($"c{character.Id}. {character.Name}: {character.Description}");
+        }
+
+        StringBuilder sb2 = new StringBuilder();
+        foreach (var voiceType in _soundGenerationSelectorService.Models)
+        {
+            sb2.Append($"{voiceType.Name},");
+        }
+
+        return $$"""
+            Assign the most fitting voice type to each character from the provided list, ensuring the chosen voice aligns with their role and attributes in the story. Only select from the available voice types.
+
+            Characters:
+            {{sb}}
+
+            Voice Types:
+            {{sb2}}
```

Comment:
string.Joinを使ってもいいかも
```suggestion
        return $$"""
            Assign the most fitting voice type to each character from the provided list, ensuring the chosen voice aligns with their role and attributes in the story. Only select from the available voice types.

            Characters:
            {{characterList.Aggregate(new StringBuilder(),
                (builder, character) => builder.AppendLine($"c{character.Id}. {character.Name}: {character.Description}"))}}

            Voice Types:
            {{_soundGenerationSelectorService.Models.Aggregate(new StringBuilder(),
                (builder, voiceType) => builder.AppendLine($"{voiceType.Name},"))}}
```